### DISCUSSION TITLE
No use of parameter `ctx`

### DISCUSSION THREAD

**Discussion by tahseenadit** [pallets/click#2722] [Q&A]
Location: https://github.com/pallets/click/blob/9f63c3b477d444245b0cc21ac23c28f6e8f4c385/src/click/core.py#L1678C5-L1682C43

```
def get_command(self, ctx: Context, cmd_name: str) -> Command | None:
        """Given a context and a command name, this returns a :class:`Command`
        object if it exists or returns ``None``.
        """
        return self.commands.get(cmd_name)
```

I was wondering why `ctx` exists as a parameter in the `get_command` function when it is not used inside the function. Do you think it should be removed ?

**Comment by davidism** [comment]
If you need information from the context when implementing a custom `get_command`, there it is.

**Reply by tahseenadit** [reply]
Thanks a lot. The response was very fast which was amazing. Do you think it also facilitates to write something like below?

```
def check_something_in_ctx():
    def wrapper(*args, **kwargs):
        ctx = args[0]
        ... # Placeholder for logics to do something with ctx
        return get_command(*args) # Assuming get_command is the original function
    return wrapper
 
@check_something_in_ctx
def custom_get_command(ctx, cmd_name):
    ... # Placeholder for logics to do something
    return 1
```

Would love to get your valuable opinion about it. Thanks!
